includes/class-wc-order.php

```diff
@@ -1449,8 +1449,8 @@ public function get_downloadable_items() {
 	 * @return bool
 	 */
 	public function needs_payment() {
-		$valid_order_statuses = apply_filters( 'woocommerce_valid_order_statuses_for_payment', array( 'pending', 'failed' ), $this );
-		return apply_filters( 'woocommerce_order_needs_payment', ( $this->has_status( $valid_order_statuses ) && $this->get_total() > 0 ), $this, $valid_order_statuses );
+		$valid_payment_statuses = apply_filters( 'woocommerce_valid_order_statuses_for_payment', array( 'unpaid', 'awaiting-authentication' ), $this );
```

Comment:
Could this filter be deprecated and replaced with a new filter so that `'pending'` / `'failed'` could still passed to the old filter, and `'unpaid'` / `'awaiting-authentication'` added to the new filter?

That would help maintain backward compatibility for 3rd party code attaching to `'woocommerce_valid_order_statuses_for_payment'` and checking for those old statuses.

Reply:
Thanks for the feedback @thenbrent, I addressed it in 75daa70 and 4f5413c

I will keep this in mind with all the other filters I touch and make the same changes accordingly.